### DISCUSSION TITLE
Did wfc works for more that 1 sensor?

### DISCUSSION THREAD

**Discussion by ds0902** [PeterMakus/SeisMIC#75] [Q&A]
Hi, at the moment I'm working in a noise data study for the OVSICORI in Costa Rica the goal of the project is create a imagen of the dv/v of a 20km zone, by using the cross-correlations of 73 sensor, but I have this problem, when in want to use the function wfc.py to get the npz files the program did not create this files, after trying so many times, i notice that even if i have two or three stations, the wfc only generate this npz data just for the station individually but no for the combinations, is this suppose to work like that? or there's some parameter or change that i need to do to get it work, I'm taking the tutorial and the params.example like a guide for getting this results, is my first time using this program.
![image_2024-10-11_131836709](https://github.com/user-attachments/assets/aaac89c5-daba-4207-bdcf-4c42a23a55f7)
Those are the correlations created by all combinations. 
![image_2024-10-11_131933241](https://github.com/user-attachments/assets/80b6602f-ae75-4e9f-b027-593745f96731)
And this are the npz created, those only include the C0005-C0005 and C0007-C0007, not the combination of them. Thank you for the help!

**Comment by PeterMakus** [comment]
Hi, thanks for the message!

To me, this seems like you have actually never computed any cross-correlations between stations but only between the components of the 3-component sensors.

In order to compute cross-correlations, you will have to set the *combination_method* parameter to *betweenStations* and then recompute the correlations (correlate.py or using the corresponding method of the Correlator object).

**Reply by ds0902** [reply]
Hi, I already try using the betweenStations in the params in order to get results, but when there come the wfc i get the same problem. for example this try 
![image_2024-10-11_142216288](https://github.com/user-attachments/assets/9ec57181-5ae9-4b84-8373-fa48b84b0a55)
Those are the initial params, there is the between stations.
![image_2024-10-11_142250756](https://github.com/user-attachments/assets/62854002-ede7-48c4-b9dd-e1abd33780ee)
and i get the cross-correlation of those 2 sensors.
and after apply the wfc function this is what i got
![image_2024-10-11_144257176](https://github.com/user-attachments/assets/09084040-a948-4b11-8468-ef7f795e793a)
There's not wfc folder.

This is another example just with one station and this what i got after use the wfc
![image](https://github.com/user-attachments/assets/b9af7390-f9b1-4f79-8cd3-184ad20bcfcf)
 
Is there anything that I have not declare like a variable or something? I check any case that the params and folder were right before apply any script?

**Comment by PeterMakus** [comment]
Hi!
That sounds like an error occurs before the output folder is even created.

Can you send the ``log_level`` in the *params.yaml* file to ``DEBUG`` and send me the output when you try to execute the waveform_coherence computation?
Right now, I still have very little to work with. It would also be useful to know: What operating system you are using, which version of SeisMIC you are on, etc. If you are using a conda environment, the output of ``conda list`` would also be handy.

**Comment by ds0902** [comment]
[conda list.txt](https://github.com/user-attachments/files/17366657/conda.list.txt)
[paramsprueba.txt](https://github.com/user-attachments/files/17366658/paramsprueba.txt)
[params2024-10-11-20-37.txt](https://github.com/user-attachments/files/17366693/params2024-10-11-20-37.txt)
There is the conda list, "paramsprueba" are the params that i declare in that example that i send to you (The one that no create the wfc folder), the other file is the params file that is created in the log folder. 
After I apply the wfc function this is the output
![image](https://github.com/user-attachments/assets/a8eee696-c73f-42a3-aebc-b49f0c5eed1b)
I understand that this first outputs are related to the corr_0.03_0.1 the correlation between this frequencies and this is done for all the frequencies that I declare.
After that at the end of the output, when is suppose to create the npz files I get this:
![image](https://github.com/user-attachments/assets/e6926077-1ce7-463c-8208-7b40c1549fd0)
I know is different because when I apply the wfc for 1 sensor this is the end of the wfc output
![image](https://github.com/user-attachments/assets/8de5817e-4d16-42f0-9d14-e50e1b84249b)

I'm working in the 0.5.24 version of seisMIC, I use Windows 10, home version and I use a Jupyter notebook in the same conda enviroment.

**Comment by PeterMakus** [comment]
Ok, Thank you for the detailled documentation!

I think I know what is happening and (if I'm right) it is indeed a bug, that I will have to fix:
The waveform coherence is computed successfully but it does not manage to compute an average of the components. However, it still deletes the separate components.

Can you re-run the wfc script but this time with the ``save_comps: True``(in the *params.yaml*) and let me know whether this works?

Thanks
Peter

**Reply by ds0902** [reply]
Hi!, I re-run the script changing the save_comps but sadly is not working. 
I get the same output at the end and the wfc folder is not created.

I'll keep looking for any other change and try that you want to do!
and thank you for all the help that you are giving me

**Comment by PeterMakus** [comment]
Hey thanks for testing this.

I will try to reproduce the error as soon I have time. I will come back to you after collecting some insights!

Peter

**Comment by ds0902** [comment]
Hey, had you the time to test something about this? I can try to do some changes if they can help you

**Comment by PeterMakus** [comment]
Hi,
Sorry for the long silence, I was very swamped with other things and had to neglect this a little.

So I tried to reproduce the failure you are getting. However, I did not find any issues and got all the expected output.

Here is what I did (step by step):

Because I am a little lazy, I just used the tutorial as starting point.

1. Just to be sure I deleted the `data` folder in the examples directory.
2. I downloaded data for X9.IR1.HHE and X9.IR3.HHE for the same time period that the tutorial is in. You can do this by modifying the first cell in the Jupyter notebook (easiest is probably to execute it twice, once with `station='IR1'`and once with `station='IR3'`.
3. I modified the `params.yaml`file *in the examples folder* (make sure to use the right parameter file). The following modifications were introduced: (1) In the **net** section, changed the `station`parameter to `'*'` (take all locally available stations). (2) In the **co** section, changed `combination_method` to `'betweenStations'`.
4. Now I executed the `wfc.py`file in the examples folder *using the terminal*. That is, you open your windows-shell, activate the right conda environment and type *python wfc.py* (make sure you are in the right directory). That should be it.


Could you try and follow exactly those steps and let me know whether this works? If not, please provide possible error messages and describe the unexpected behaviour.


Best,
Peter
